apps/api/src/app/notification-template/usecases/delete-notification-template/delete-notification-template.usecase.ts

```diff
@@ -0,0 +1,33 @@
+// eslint-ignore max-len
+
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { NotificationTemplateEntity, NotificationTemplateRepository } from '@novu/dal';
+
+import { GetNotificationTemplateCommand } from '../get-notification-template/get-notification-template.command';
+
+@Injectable()
+export class DeleteNotificationTemplate {
+  constructor(private notificationTemplateRepository: NotificationTemplateRepository) {}
+
+  async execute(command: GetNotificationTemplateCommand): Promise<NotificationTemplateEntity> {
+    const existingTemplate = await this.notificationTemplateRepository.findById(
+      command.templateId,
+      command.organizationId
+    );
+    if (!existingTemplate) throw new NotFoundException(`Entity with id ${command.templateId} not found`);
+
+    await this.notificationTemplateRepository.update(
+      {
+        _id: command.templateId,
+        _organizationId: command.organizationId,
+      },
+      {
+        $set: {
+          isDeleted: true,
+        },
```

Comment:
Currently, we use 'mongoose-delete' on several repositories, it helps us to manage the variable such as isDeleted, deletedAt, and deletedBy. I think we could use it here as well, what do you think? 
For further examples, you can look into the integration repository and its use there.

Reply:
I had a discussion with @scopsy yesterday and I started copy-pasting that... Smart Coder :)